tests/ui/typeck/issue-115264-pat-field.rs

```diff
@@ -0,0 +1,19 @@
+// Regression test for issue 115264
+// Tests that retrieving the ident of 'foo' variable in
+// the pattern inside main() does not cause an ICE
+
+// check-pass
+
+struct X {
+    foo: i32,
+}
+
+#[allow(unused_variables)]
+fn main() {
+    let X {
+        #[doc(alias = "StructItem")]
```

Comment:
This should generate the same warning as something like:

```
fn main() {
    #[doc(alias = "test")]
    let x = 1;
    //~^ WARN rustdoc does not generate documentation for statements
}

Reply:
Can be done as a follow-up though.

Reply:
Sure. Will do it in a separate PR

Reply:
@compiler-errors I've created an issue for this: https://github.com/rust-lang/rust/issues/115462